src/arch/x86/bios/intelmp.cc

```diff
@@ -85,7 +85,7 @@ writeOutField(PortProxy& proxy, Addr addr, T val)
 uint8_t
 writeOutString(PortProxy& proxy, Addr addr, std::string str, int length)
 {
-    char cleanedString[length + 1];
+    auto cleanedString = static_cast<char *>(malloc(length + 1));
```

Comment:
This is a small thing, but could we use a string? Or, maybe better, a `std::vector<char>`?

I'd rather use C++ things and not malloc/free, but it's not a big deal either way.

Reply:
ACK. Done.